src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +207,167 @@ async function uploadWithProgress({
+            ignoreFocusOut: true,
+            title: localize('AWS.message.selectBucket','Select an S3 bucket to upload to'),
+            step: 2,
+            totalSteps: 2,
+        },
+        items: [...bucketItems, createNewBucket],
+        buttons: [vscode.QuickInputButtons.Back]
+    })
+    const response = verifySinglePickerOutput(await promptUserFunction({
+        picker: picker,
+        onDidTriggerButton: (button, resolve, reject) => {
+            if (button === vscode.QuickInputButtons.Back) {
+                resolve([{
+                    label: 'back',
+                    bucket: undefined
+                }])
+            }
+
+        },
+    })) 
+
+    if (!response) {
+        return 'cancel'
+    }
+
+    if (!response.bucket) {
+        if (response.label === 'back') {
+            return response.label
+        }
+        if (response.label === 'Create new bucket') {
+            const s3Node = new S3Node(s3client)
+            await createBucket(s3Node, window, Commands.vscode())
+            return promptUserForBucket(s3client)
+        }
+    } else {
+        return response.bucket
+    }
+    return 'cancel'
+}
+
+/**
+ * Gets the open file in the current editor
+ * Asks the user to browse for more files
+ * If no file is open it prompts the user to select file
+ * @param document document to use as currently open
+ * 
+ * @returns file selected by the user
+ */
+export async function getFileToUpload(
+    document?: vscode.Uri,
+    window = Window.vscode(),
+    promptUserFunction = promptUser
+): Promise<vscode.Uri| undefined> {
+    
+    let fileLocation: vscode.Uri | undefined
+
+    if (!document) {
+        fileLocation = await promptForFileLocation(window)
+    } 
+    else {
+        fileLocation = document
+        const fileNameToDisplay = path.basename(fileLocation.fsPath)
+
+        const fileOption : vscode.QuickPickItem = { 
+            label: addCodiconToString('file', fileNameToDisplay) 
+        }
+        const selectMore: vscode.QuickPickItem = {
+            label: 'Browse for more files...'
+        }
+
+        const picker = createQuickPick({
+            options: {
+                canPickMany: false,
+                ignoreFocusOut: true,
+                title: localize('AWS.message.selectFileUpload', 'Select a file to upload'), 
+                step: 1,
+                totalSteps: 2,
+            },
+            items: [fileOption, selectMore],
+            buttons: [vscode.QuickInputButtons.Back]
+        })
+
+        
+        const response = verifySinglePickerOutput(await promptUserFunction({
+            picker: picker,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                }
+            },
+        }))
+
+        if(!response){
+            return
+        }
+
+        if (response?.label === 'Browse for more files...') {
```

Comment:
You might be able to use `response === selectMore`. At the very least, `response.label === selectMore.label`. This becomes more important since you should localize the title.